### PR TITLE
Add types for Application and Response

### PR DESCRIPTION
**Pull request by seedofjoy**
## What do these changes do?
Add types for Application and Response

## Are there changes in behavior for the user?
No

## Related issue number
WIP on #1749

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: aiohttp/web_app.py [hunk 1]
```diff
@@ -39,13 +54,14 @@ class Application(MutableMapping):
...
-                 logger=web_logger,
-                 router=None,
-                 middlewares=(),
-                 handler_args=None,
-                 client_max_size=1024**2,
-                 loop=None,
-                 debug=...):
+                 logger: logging.Logger=web_logger,
+                 router: Optional[UrlDispatcher]=None,
+                 middlewares: Sequence[_Middleware]=(),
```
[the remark this answers is not in the document]
Comment (asvetlov) [reply aiohttp/web_app.py:59]: Let's keep sequence. We have no explicit test for `None`, I don't want to keep the guarantee forever.
BTW `0` and `0.0` should work too but it is weird.

File: aiohttp/web_response.py [hunk 2]
```diff
@@ -438,16 +443,23 @@ def __eq__(self, other):
...
+                 charset: Optional[str]=None) -> None:
         if body is not None and text is not None:
             raise ValueError("body and text are not allowed together")
 
         if headers is None:
             headers = CIMultiDict()
         elif not isinstance(headers, (CIMultiDict, CIMultiDictProxy)):
             headers = CIMultiDict(headers)
+        else:
+            headers = cast(CIMultiDict, headers)
```
[the remark this answers is not in the document]
Comment (asvetlov) [reply aiohttp/web_response.py:462]: Let's keep the line as is

**Comment by asvetlov** [comment]
Please make tests passing on Python 3.5

**Comment by seedofjoy** [comment]
> Please make tests passing on Python 3.5

done

**Comment by asvetlov** [comment]
thanks

**Pull request merged**